### DISCUSSION TITLE
:ordered and :unordered task lists

### DISCUSSION THREAD

**Discussion by ko56** [shop-planner/shop3#129] [Q&A] [answered]
In reading sec. 4.9 of the manual, I realized that I am not clear whether a task list is ordered or unordered by default,  i.e. when neither :ordered nor :unordered is specified. The fact that :unordered has to be specified explicitly whereas :ordered is optional seems to indicate that :ordered is the default.  Is this correct?

**Comment by rpgoldman** [comment, marked as answer]
Yes, that is correct.

Honestly, I regret the optionality of `:ordered` -- it makes parsing task networks much more complex and error-prone.  There's a lot of "look at the first element and if it's a list ..." kind of stuff.j

**Reply by ko56** [reply]
Thanks.  I suggest that this be clarified in the manual.
Since I don't have a lot of SHOP3 code to maintain, I would not be opposed
to making it mandatory :-)
